### DISCUSSION TITLE
SAP and Diesel

### DISCUSSION THREAD

**Discussion by brunbjerg** [diesel-rs/diesel#4048] [Ideas]
Hello Diesel Community,

Does anyone of you in here have experience with using Rust together with the SAP relational database? 

I have an application written in Rust and having the integration SAP through Diesel would be very beneficial

Kind regards,
Christian

**Comment by weiznich** [comment]
I personally do not have any experience with SAP. From a quick internet search it seems like they provide a SQL interface but you need to use their libraries/API's to program against that interface. In this case this would be very similar to all the other commercial/closed source database that diesel does not support out of the box: You would need to write your own connection/backend implementation for this as we cannot support that in diesel itself. See the [documentation](https://docs.diesel.rs/2.1.x/diesel/connection/trait.Connection.html#implementing-a-custom-connection) for details on how to approach that problem.

**Reply by brunbjerg** [reply]
Thank you for the reply. I will write the necessary extensions myself to interact with SAP.

Just to be sure, there is no interest from diesel, to have such SAP code as part of the crate?

SAP applications often have to be extremely reliable, and coding modern extensions in Rust would be a good fit as operational companies that SAP target, are afraid of long tail risks associated buggy software

**Reply by weiznich** [reply]
> Just to be sure, there is no interest from diesel, to have such SAP code as part of the crate?

The general policiy for new backends in diesel is:

* The target database needs to be open source so that's easy to test for contributors
* The target database needs to be widely used
* There needs to be at least one maintainer that's familiar with the target database.

As far as I know the first point is not fulfilled by SAP. The last point is currently also not fulfilled.
